python/Ganga/GPIDev/Base/Proxy.py

```diff
@@ -229,24 +244,36 @@ def isType(_obj, type_or_seq):
         return isinstance(obj, bare_type_or_seq)
 
 def getName(_obj):
-    """Strip any proxy and then return an objects name"""
+    """Strip any proxy and then return an objects name
+    Args:
+        _obj (object): This may be an instance or a class"""
     obj = stripProxy(_obj)
     returnable = _getName(obj)
     return returnable
 
 def stripProxy(obj):
-    """Removes the proxy if there is one"""
+    """Removes the proxy if there is one
+    Args:
+        obj (object): This may be an instance or a class
+    """
     return getattr(obj, implRef, obj)
 
 
 def addProxy(obj):
-    """Adds a proxy to a GangaObject"""
+    """Adds a proxy to a GangaObject instance or class
+    Args:
+        obj (GangaObject, class): This may be a Ganga object or class which you're wanting to add a proxy to
```

Comment:
When I made the suggestion in the SharedFile PR to use `addProxy()` on a class name to get the proxied version I didn't realise it didn't currently support it. I'd rather keep this function simple and only use it on instances, not classes. If you need to get the proxy class version of a `GangaObject` then there's `getProxyClass()`. There's no need for a single function to be able to cover all bases, different situations need different solutions.